Add test to validate that docker functionality is supported

dylan-matthew-turner opened the pull request:
We want to support docker, so we need to ensure that things like the daemon, `--network host`, parallel containers, etc all work.

WI #2500898

This PR provides two tests:

- test_daemon which simply verifies that `docker run hello-world` runs without error
- test_parallel which sets up a background docker container with a file server and a second docker container which reads the file from that server

Tested by building the package and running ptest-runner on a 9053

amstewart commented on recipes-ni/docker-functional-tests/docker-functional-tests.bb:
```diff
@@ -0,0 +1,37 @@
+
+inherit ptest
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
```
The `FILESEXTRAPATHS` trick is only needed in `.bbappend` files. The recipe-local `files/` directory should already be in the PATHS for a `.bb` recipe.

amstewart commented on recipes-ni/docker-functional-tests/docker-functional-tests.bb:
```diff
@@ -0,0 +1,37 @@
+LICENSE = "GPL-2.0-only & GPL-2.0-or-later"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/GPL-2.0-or-later;md5=fed54355545ffd980b814dab4a3b312c"
+
+inherit ptest
```
Try to follow the recipe layout [from the styleguide](https://github.com/ni/meta-nilrt/blob/nilrt/master/kirkstone/docs/styleguide.md#layout). Class inherits should be pushed down toward the middle of the file.

amstewart commented on recipes-ni/docker-functional-tests/docker-functional-tests.bb:
```diff
@@ -0,0 +1,37 @@
+
+ALLOW_EMPTY:${PN} = "1"
+
+SRC_URI += "\
```
MINOR: since this is a bb-recipe, we should prefer to use the regular `=` assignment, to keep anything screwy from happening with the `SRC_URI` variable.

amstewart commented on recipes-ni/docker-functional-tests/docker-functional-tests.bb:
```diff
@@ -0,0 +1,37 @@
+SUMMARY = "Functional test of docker container capabilities on NILRT"
+HOMEPAGE = "https://docker.com"
```
For NI-authored recipes, we generally set the `HOMEPAGE` to `https://github.com/ni/nilrt`.

amstewart commented on recipes-ni/docker-functional-tests/docker-functional-tests.bb:
```diff
@@ -0,0 +1,37 @@
+SUMMARY = "Functional test of docker container capabilities on NILRT"
+HOMEPAGE = "https://docker.com"
+SECTION = "tests"
+LICENSE = "GPL-2.0-only & GPL-2.0-or-later"
```
Why not just MIT?

dylan-matthew-turner replied:
I copied over from another test that I used as reference. Is MIT our preference for new recipes?

amstewart replied:
I'd say so; yeah.

chaitu236 reviewed (changes requested):
Can you add work item number to description

dylan-matthew-turner committed:
docker-func-tests: test to validate docker support

Includes 2 tests:

- test_daemon which just ensures docker can run
- test_parallel which ensures two parallel containers can communicate

Signed-off-by: Dylan Turner <dylan.turner@ni.com>

dylan-matthew-turner commented:
Changes from previous commits:

- restructured to fit layout guide
- switched from GPL2 to MIT license
- remove unneeded changes to work with multiple recipes
- correct homepage
